### ISSUE TITLE
New vector math operations needed

### ISSUE THREAD

**Issue by LeonCheung** [JacquesLucke/animation_nodes#346]
A few more simple operatiions like **Absolute** and **Round** would make liife a bit easier for some cases. (well, expression really needs quite a bit more work each time...)

**Comment by JacquesLucke** [comment]
For what do you need these two operations?
I mean they aren't too hard to implement but I don't see how they can be useful..

**Comment by LeonCheung** [comment]
For Absolute, please see the attached file in my last post on BA [here](http://blenderartists.org/forum/showthread.php?350296-Addon-Animation-Nodes&p=2974014&viewfull=1#post2974014).

You can see that I want to take scale vector from other objects, then use it on a text object. However, when the object have minus scale value, the text will be flipped. To make the text "readable", an absolute scale would be used for it.

Basically, it can be useful to limit the target objects' vectors within a positive range (or negative).

For Round, It can be helpful to snap vectors to the Blender world grid. or to generate some positions (or rotations, scales) with certain level of imcrements.

**Comment by JacquesLucke** [comment]
The changes I make to the vector math node will only be available in the AN 1.5 branch.

**Comment by JacquesLucke** [comment]
How should the input for a *round* operation look like?
The amount of decimals wouldn't help in most cases, right?
More something like "Step Size". And maybe we shouldn't call it "Snap Vector" instead of "Round Vector".
What do you think?

**Comment by LeonCheung** [comment]
I've no idea on how this step concept works, just brainstorming.  :D 

BTW, It appears a bit silly in the way below:
![round_location](https://cloud.githubusercontent.com/assets/494788/11456508/8386fd40-96c6-11e5-8f69-9f4af058c1cb.png)

**Comment by LeonCheung** [comment]
Ah, I think I see what you mean by step, Truely useful! Sort of "Snap to Step/Increment"?

**Comment by LeonCheung** [comment]
Something like this?
![euler step](https://cloud.githubusercontent.com/assets/494788/11456718/9cf1d452-96cc-11e5-874b-0802f1b05ead.png)

**Comment by JacquesLucke** [comment]
Yes a bit like this. Just for normal vectors first :)

**Comment by og76** [comment]
see also #327 
And why just the vectors first? The variant I was making is a "data rounding" with opt for vec/eul/quat
or, cause in this case the formula is slightly different, should go into math ?

**Comment by JacquesLucke** [comment]
![snap vector](https://cloud.githubusercontent.com/assets/5881326/11631253/795f2f72-9d01-11e5-8628-de17f315729f.PNG)
